.eslintrc

```diff
@@ -0,0 +1,29 @@
+{
+  // tells eslint to use the TypeScript parser
+  "parser": "@typescript-eslint/parser",
+  // tell the TypeScript parser that we want to use JSX syntax
+  "parserOptions": {
+    "tsx": true,
+    "jsx": true,
+    "useJSXTextNode": true,
+    "project": "./tsconfig.json",
+    "tsconfigRootDir": "."
+  },
+  // we want to use the recommended rules provided from the typescript plugin
+  "extends": [
+    "plugin:@typescript-eslint/recommended",
+    "plugin:patternfly-react/recommended"
+  ],
+  // includes the typescript specific rules found here: https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#supported-rules
+  "plugins": ["@typescript-eslint", "react-hooks", "eslint-plugin-react-hooks"],
+  "rules": {
+    "@typescript-eslint/explicit-function-return-type": "off",
+    "react-hooks/rules-of-hooks": "error",
+    "react-hooks/exhaustive-deps": "warn",
+    "@typescript-eslint/interface-name-prefix": "off",
+    "prettier/prettier": "off",
```

Comment:
I'm with ya. Maybe, for now, we leave as an option that users can toggle on their own. If we can come to some consensus on what rules to apply I'm fine with turning it on for the masses - I just haven't so far because it seems like a pretty controversial topic for some 😉

The good thing is, this PR brings us one step closer to having support in this area, so if a user really wants it - should just be a few minutes of swizzling the config to make it happen.